iidm/iidm-api/src/main/java/com/powsybl/iidm/network/IdBasedBusRef.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.powsybl.commons.PowsyblException;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * There would be two types of id:
+ * 1. id of equipment:
+ * 2. id of a configured bus itself:
+ * @author Yichen TANG <yichen.tang at rte-france.com>
+ */
+public class IdBasedBusRef implements BusRef {
+
+    private final String id;
+
+    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
+    public IdBasedBusRef(@JsonProperty("id") String id) {
+        this.id = Objects.requireNonNull(id);
+    }
+
+    @Override
+    public Optional<Bus> resolve(Network network) {
+        final Identifiable<?> identifiable = network.getIdentifiable(id);
+        if (identifiable == null) {
+            return Optional.empty();
+        }
+        if (identifiable instanceof Bus) {
+            String busId = identifiable.getId();
+            return Optional.of(network.getBusView().getBus(busId));
```

Comment:
In that case it is probably better to use the voltage level bus view, it will restrict the search.

Otherwise, for example In iidm-impl, the search will go over the list of all voltage levels.

Reply:
**Second thought:**
actually it's not what we want to do. We need to return the bus (of the bus view) which contains that identifiable bus !

I think there is no easy way for this in the API today (it's probably a missing feature). As a workaround we could use terminals : return the first bus which contains one of the terminals of the identifiable bus. In the weird case where there is no terminal connected to the identifiable bus, return empty.